### PR TITLE
Adds a progress bar to chapters

### PR DESCRIPTION
**Pull request by duarte**
Adds a progress bar to the chapters top border

### CODE REVIEW

File: src/css/_docs.scss [hunk 1]
```diff
@@ -481,12 +482,25 @@
...
+    }
   }
   &-active {
     //Apologies but can only make work with !important
-    border-top: 1px solid $c-grey-lightest !important;
+    border-top: 1px solid $c-grey-light !important;
     opacity: 1;
     transition: all 360ms;
+    .progress {
+      opacity: 1 !important;
```
Comment (akash1810) [inline src/css/_docs.scss]: `!important` are used as a last resort css 😄 can we add a comment explaining why the `!important` is used?

File: src/js/lib/youtube.js [hunk 2]
```diff
@@ -21,11 +21,13 @@ export function pimpYouTubePlayer(videoId, node, height, width, chapters) {
 
                             const playerTotalTime = youTubePlayer.getDuration();
                             chapTimer = setInterval(function() {
+                                let chapterCurrentProgress;
```
Comment (akash1810) [inline src/js/lib/youtube.js]: why is this declared so high up?

File: src/js/lib/youtube.js [hunk 3]
```diff
@@ -35,6 +37,8 @@ export function pimpYouTubePlayer(videoId, node, height, width, chapters) {
                                         if (el.dataset.sheetTimestamp === currentChapter[0].chapterTimestamp){
                                             el.classList.add('docs--chapters-active');
                                             el.classList.remove('docs--chapters-inactive');
+                                            const progress = el.querySelector('.progress');
+                                            progress.style.width = chapterCurrentProgress*100+'%';
```
Comment (akash1810) [inline src/js/lib/youtube.js]: can use es6 string here for added niceness 😄

File: src/js/lib/youtube.js (hunk 2 above)
Comment (duarte) [inline src/js/lib/youtube.js]: I think it's the latest I can declare it — needs to be accessible in lines 30 and 41.
Is there another place I could declare it?

**Comment by akash1810** [comment]
so every chapter has a progress bar?

can we add a screenshot of this in action?

do we need to turn this off on mobile?

File: src/js/lib/youtube.js (hunk 3 above)
Comment (duarte) [inline src/js/lib/youtube.js]: Done! :)

**Comment by duarte** [comment]
Screenshot of this in action
screen shot 2016-09-08 at 12 04 40

File: src/js/lib/youtube.js [hunk 4]
```diff
@@ -21,11 +21,13 @@ export function pimpYouTubePlayer(videoId, node, height, width, chapters) {
 
                             const playerTotalTime = youTubePlayer.getDuration();
                             chapTimer = setInterval(function() {
+                                let chapterCurrentProgress;
                                 const playerCurrentTime = youTubePlayer.getCurrentTime();
                                 const currentChapter = chapters.filter(function(value){
                                     const chapStart = value.chapterTimestamp;
                                     const chapNext = value.nextChapter || playerTotalTime;
                                     if(playerCurrentTime >= chapStart && playerCurrentTime <= chapNext){
+                                        chapterCurrentProgress = (playerCurrentTime-chapStart)/(chapNext-chapStart);
```
Comment (akash1810) [inline src/js/lib/youtube.js]: not sure if this should be done within the `filter` method?

File: src/js/lib/youtube.js (hunk 2 above)
Comment (akash1810) [inline src/js/lib/youtube.js]: > Is there another place I could declare it?

I'd move `chapterCurrentProgress = (playerCurrentTime-chapStart)/(chapNext-chapStart);` out from the `filter` method and do this work inside the `forEach` later on.

I'd say `filter` should only be responsible for finding a value in an array make things easier to reason about.

Comment (duarte) [inline src/js/lib/youtube.js]: Would I then need to declare chapStart, chapNext outside the filter?

**Comment by akash1810** [comment]
> I think it's still a valid pattern / useful in mobile, so if it's not problematic I'd like to keep it on…

On mobile, you don't see all the chapters in the viewport. Presumably to keep this feature in view, the `ul` will have to scroll right as the video progresses to bring the later chapters into the viewport?

**Comment by mr-mr** [comment]
The scrolling on mobile is already built in.

File: src/js/lib/youtube.js (hunk 2 above)
Comment (akash1810) [inline src/js/lib/youtube.js]: you'd need to re-declare them, yep

**Comment by akash1810** [comment]
👍

**Pull request merged**